### ISSUE TITLE
Client can't duplicate forms

### ISSUE THREAD

**Issue by darylknight** [solspace/craft-freeform#434]
**Describe the bug or issue you're experiencing**

I can replicate an issue a client has just asked about - they are unable to duplicate existing forms. I can, but when they try, they get a server error (tested in both Craft 3 and Craft 4, as I've just updated the site to Craft 4 to see if that fixed it).

The server error is:

```
2022-08-22 10:42:42 [web.ERROR] [yii\db\IntegrityException] SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'permissionId' cannot be null The SQL being executed was: INSERT INTO `userpermissions_usergroups` (`groupId`, `permissionId`, `dateCreated`, `dateUpdated`, `uid`) VALUES (3, NULL, '2022-08-22 10:42:42', '2022-08-22 10:42:42', 'ce3c4225-67d2-44a0-8a02-4bbb07dda8f0') {"trace":["#0 /home/[removed]-qbaxe/[removed].com/vendor/yiisoft/yii2/db/Command.php(1307): yii\\db\\Schema->convertException()","#1 /home/[removed]-qbaxe/[removed].com/vendor/yiisoft/yii2/db/Command.php(1102): yii\\db\\Command->internalExecute()","#2 /home/[removed]-qbaxe/[removed].com/vendor/yiisoft/yii2/db/Schema.php(431): yii\\db\\Command->execute()","#3 /home/[removed]-qbaxe/[removed].com/vendor/yiisoft/yii2/db/ActiveRecord.php(604): yii\\db\\Schema->insert()","#4 /home/[removed]-qbaxe/[removed].com/vendor/yiisoft/yii2/db/ActiveRecord.php(570): yii\\db\\ActiveRecord->insertInternal()","#5 /home/[removed]-qbaxe/[removed].com/vendor/yiisoft/yii2/db/BaseActiveRecord.php(676): yii\\db\\ActiveRecord->insert()","#6 /home/[removed]-qbaxe/[removed].com/vendor/solspace/craft-freeform/packages/plugin/src/Controllers/FormsController.php(376): yii\\db\\BaseActiveRecord->save()","#7 /home/[removed]-qbaxe/[removed].com/vendor/solspace/craft-freeform/packages/plugin/src/Controllers/FormsController.php(136): Solspace\\Freeform\\Controllers\\FormsController->updateGroupPermissions()","#8 [internal function]: Solspace\\Freeform\\Controllers\\FormsController->actionDuplicate()","#9 /home/[removed]-qbaxe/[removed].com/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array()","#10 /home/[removed]-qbaxe/[removed].com/vendor/yiisoft/yii2/base/Controller.php(178): yii\\base\\InlineAction->runWithParams()","#11 /home/[removed]-qbaxe/[removed].com/vendor/yiisoft/yii2/base/Module.php(552): yii\\base\\Controller->runAction()","#12 /home/[removed]-qbaxe/[removed].com/vendor/craftcms/cms/src/web/Application.php(301): yii\\base\\Module->runAction()","#13 /home/[removed]-qbaxe/[removed].com/vendor/yiisoft/yii2/web/Application.php(103): craft\\web\\Application->runAction()","#14 /home/[removed]-qbaxe/[removed].com/vendor/craftcms/cms/src/web/Application.php(286): yii\\web\\Application->handleRequest()","#15 /home/[removed]-qbaxe/[removed].com/vendor/yiisoft/yii2/base/Application.php(384): craft\\web\\Application->handleRequest()","#16 /home/[removed]-qbaxe/[removed].com/public/index.php(12): yii\\base\\Application->run()","#17 {main}"],"memory":14714928,"exception":"[object] (yii\\db\\IntegrityException(code: 23000): SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'permissionId' cannot be null\nThe SQL being executed was: INSERT INTO `userpermissions_usergroups` (`groupId`, `permissionId`, `dateCreated`, `dateUpdated`, `uid`) VALUES (3, NULL, '2022-08-22 10:42:42', '2022-08-22 10:42:42', 'ce3c4225-67d2-44a0-8a02-4bbb07dda8f0') at /home/[removed]-qbaxe/[removed].com/vendor/yiisoft/yii2/db/Schema.php:676)\n[previous exception] [object] (PDOException(code: 23000): SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'permissionId' cannot be null at /home/[removed]-qbaxe/[removed].com/vendor/yiisoft/yii2/db/Command.php:1302)"} 
```

The user permissions for the client are:

![image](https://user-images.githubusercontent.com/1250017/185903818-c5237b30-2a75-419f-b2f9-a71c137d42b6.png)

**Steps to reproduce**

1. Login as a non-developer account
2. Try to duplicate a form

**Expected behavior**
Clients should be able to duplicate forms with those permissions

**Craft & Plugin Info (please complete the following information):**

- Craft Version: 4.2.1.1
- Freeform Version: 4.0.0
- Freeform Edition: Pro
- Fresh Install or Upgrade: Upgrade

*Ref: CRD*

**Comment by darylknight** [comment]
I'm also getting this from the Dasboard. I'm not aware of a duplicate option within the form builder, but "save as a new form" works fine.

**Comment by kjmartens** [comment]
This is now resolved in Freeform 3.13.19 and 4.0.3. 🙂
